// pages/articleCreate/articleCreate.js
import React, {useContext, useEffect, useState} from 'react';
import ArticleForm from 'components/articleForm';
import useFetch from 'hooks/useFetch';
import {Redirect} from 'react-router-dom';
import {CurrentUserContext} from 'contexts/currentUser';

const ArticleCreate = () => {
    const [successfulSubmit, setSuccessfulSubmit] = useState(false);
    const [currentUserState] = useContext(CurrentUserContext);
    const initialValues = {
        title: '',
        body: '',
        description: '',
        tagList: []
    };
    const apiUrl = '/articles';
    const [{response, error}, doFetch] = useFetch(apiUrl);


    const handleSubmit = (article) => {

        doFetch({
            method: 'post',
            data: {article}
        });
    };

    useEffect(() => {
        if (!response) {
            return;
        }
        setSuccessfulSubmit(true);
        console.log('response', response);
    }, [response]);

    if (currentUserState.isLoggedIn === false) {
        return <Redirect to="/"/>;
    }

    if (successfulSubmit) {
        return <Redirect to={`/articles/${response.article.slug}`}/>;
    }
    return <div>
        <ArticleForm
            errors={(error && error.errors) || {}}
            initialValues={initialValues}
            onSubmit={handleSubmit}/></div>;
};

export default ArticleCreate;
